sdk/go/common/resource/plugin/rpc.go

```diff
@@ -166,8 +166,11 @@ func MarshalPropertyValue(v resource.PropertyValue, opts MarshalOptions) (*struc
 		ref := v.ResourceReferenceValue()
 		if !opts.KeepResources {
 			val := string(ref.URN)
-			if ref.ID != "" {
+			if ref.HasID {
 				val = string(ref.ID)
+				if val == "" {
+					val = UnknownStringValue
```

Comment:
I'm not sure that this is correct--don't the SDKs assume that an empty ID is an unknown ID? Or does this "just work" in the various unmarshallers b/c they call into lower-level unmarshalling code to unmarshal the ID itself?

Reply:
Ah, I see. I'll have to check (but you're probably right).

Reply:
I do think that we'll either need to change all of the marshalers to understand the unknown property value in the ID field of a resource ref or add a `HasID` bit to the wire format. I'm slightly in favor of the former, in which case we should consider revising the in-memory representation similarly. I chose the `HasID` bit there because I wanted to retain the strongly-typed ID field (instead of a `resource.PropertyValue` that could hold a computed property).